### ISSUE TITLE
Make it possible to run invoker as fat jar with Boot-based function

### ISSUE THREAD

**Issue by trisberg** [projectriff/java-function-invoker#86]
Should be able to run a Spring Boot based function using:

```
java -jar ./target/java-function-invoker-0.0.8-exec.jar --function.uri=...
```

**Comment by dsyer** [comment]
This is caused by the fact that the fat jar ends up as the parent class loader path, because of the way that the `JarLauncher` itself works. The application classes are in `BOOT-INF/classes` so they are properly hidden, but the `META-INF/spring.factories` is in the root of the jar. We just need to adjust the parent classpath to skip the fat jar.  Opened https://github.com/spring-cloud/spring-cloud-function/issues/201.

**Comment by dsyer** [comment]
Should be fixed in 1dbaf73
